### ISSUE TITLE
Stops working with different CPU on identical Motherboard

### ISSUE THREAD

**Issue by macattackjack** [RehabMan/OS-X-USB-Inject-All#20]
My Current working configuration is:

Asus Sabertooth X79
i7-4930k @3.4 6 core
32 GB ram
Evga GeForce GT 710 2GB DDR3 gpu
960 GB Patriot SSD
10.14.3
Clover v2.4k_r4862
No DSDT file used.

Mojave 10.14.3 Installed, all USB ports working, all the time using USBInjectAll.kext v 0.7.1, with no patching needed to increase port numbers since the available number covers all of them.

However, when I use this same SSD that Mojave is installed on, in another computer with an exact same motherboard, GPUs, etc, with an E5-2697 v2 12 core CPU instead, USB 3.0 stops working. Only USB 2.0 works.

Is there is something in the USBInjectAll.kext that includes the i7-4039k CPU but doesn't include the E5-2697 v2?

I don't know that there is a difference in version numbers on the motherboards or if that could affect this.

I have confirmed all bios settings are identical.

I even tried switching all the cards in the E5-2697 v2 computer so they'd be in the same order as the i7-4930k, taking out a USB 3.0 card. Nothing solved the problem.

The only other difference is the bios version and ME versions of the board. If I remember correctly, I updated the bios to a newer version when I installed the E5-2697 v2.

I have tried various configurations with the Config file, expanding the number of USB ports, using the old SSDT from Sierra and High Sierra, different configs and EFI folders from here and from elsewhere. Nothing has solved the problem. I also tested on a Asus Rampage IV Gene, and again there were no issues. It seems to be related to the CPU, but short of taking the E5-2697 v2 out of the computer with the USB 3 issues and putting in the one that's working to see if it is the CPU specifically causing the problems, I thought I'd check here first. Posting this problem on other hackintosh sites produced no solutions as to why it would work on one computer but not the other.

GenericUSBXHCI, which has worked up to high 10.13.6, doesn't really work with Mojave on this same computer. It works initially, but if I unplug the usb device it stops working and 2.0 ports don't work at all.

Any help is appreciated.

[config.txt](https://github.com/RehabMan/OS-X-USB-Inject-All/files/2846628/config.txt)
![drivers64uefi](https://user-images.githubusercontent.com/47461050/52502562-ee533f80-2b97-11e9-9529-a32c33c8cdef.png)
![kexts-other](https://user-images.githubusercontent.com/47461050/52502572-f3b08a00-2b97-11e9-8c9f-6bbef953305d.png)

**Comment by wsy** [comment]
As you said, with the original mother board and i7, your USB is working all right.
So, instead of moving SSD to another mother board, have you tried replace i7 with E5 on the original mother board? (Although they're exactly the same model)
Maybe it's something related with BIOS version since the ACPI tables may have been changed.

**Comment by macattackjack** [comment]
I think I can confirm it's the bios version without changing the CPU. I have a third identical Sabertooth with an identical i7 4930k with the same bios as the E5, and USB 3.0 doesn't work. So

Computer 1 (Original Install computer)
Sabertooth X79 Bios version 4608 i7 4930k - all USB ports work.
Computer 2
Sabertooth X79 Bios version 4701 E5-2697 v2 - USB 3.0 ports don't work, USB 2.0 ports do work.
Computer 3
Sabertooth X79 Bios version 4701  i7 4930k - USB 3.0 ports don't work, USB 2.0 ports do work.

I don't think it's wise to backup the bios version to a previous version in the E5 computer, and I'd rather not put the E5 in the other motherboard, as it really doesn't solve the problem since I'd still have two computers that the internal USB doesn't work. While bios version 4608 does support the E5, again, going to a previous version may prove unstable, and as I believe I updated the bios on that motherboard when I upgraded to the E5 CPU.

Is there anything that can be done on the software side for this, or is it something I just have to live with? As much as I like the Asus board, this one has proved to be trouble when it comes to USB.
